ark/utils/data_utils_test.py

```diff
@@ -1,325 +1,211 @@
-        chans = [chan.split(".tiff")[0] for chan in imgs]
-        _create_img_dir(temp_dir, fovs, imgs)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        fovs, chans, imgs = test_utils.gen_fov_chan_names(3, 3, return_imgs=True)
+
+        filelocs, data_xr = test_utils.create_paired_xarray_fovs(
+            temp_dir, fovs, chans, img_shape=(10, 10), delimiter='_', fills=True, sub_dir="TIFs",
+            dtype="int16"
+        )
 
         # check default loading of all files
-        test_loaded_xr = \
+        loaded_xr = \
             data_utils.load_imgs_from_tree(temp_dir, img_sub_folder="TIFs", dtype="int16")
 
-        # make sure all folders loaded
-        assert np.array_equal(test_loaded_xr.fovs.values.sort(), fovs.sort())
-
-        # make sure all channels loaded
-        assert np.array_equal(test_loaded_xr.channels.values.sort(), chans.sort())
+        assert loaded_xr.equals(data_xr)
 
         # check loading of specific files
         some_fovs = fovs[:2]
         some_imgs = imgs[:2]
         some_chans = chans[:2]
 
-        test_subset_xr = \
+        loaded_xr = \
             data_utils.load_imgs_from_tree(temp_dir, img_sub_folder="TIFs", dtype="int16",
                                            fovs=some_fovs, channels=some_imgs)
 
-        # make sure specified folders loaded
-        assert np.array_equal(test_subset_xr.fovs.values.sort(), some_fovs.sort())
-
-        # make sure specified channels loaded
-        assert np.array_equal(test_subset_xr.channels.values.sort(), some_chans.sort())
+        assert loaded_xr.equals(data_xr[:2, :, :, :2])
 
         # check loading w/o file extension
-        test_noext_xr = \
+        loaded_xr = \
             data_utils.load_imgs_from_tree(temp_dir, img_sub_folder="TIFs", dtype="int16",
                                            channels=some_chans)
 
-        # make sure all folders loaded
-        assert np.array_equal(test_noext_xr.fovs.values.sort(), fovs.sort())
-
-        # make sure specified channels loaded
-        assert np.array_equal(test_noext_xr.channels.values.sort(), some_chans.sort())
+        assert loaded_xr.equals(data_xr[:, :, :, :2], )
 
         # check mixed extension presence
-        test_someext_xr = \
+        loaded_xr = \
             data_utils.load_imgs_from_tree(temp_dir, img_sub_folder="TIFs", dtype="int16",
                                            channels=[chans[i] if i % 2 else imgs[i]
                                                      for i in range(3)])
 
-        # make sure all folders loaded
-        assert np.array_equal(test_someext_xr.fovs.values.sort(), fovs.sort())
+        assert loaded_xr.equals(data_xr)
 
-        # makes sure all channels loaded
-        assert np.array_equal(test_someext_xr.channels.values.sort(), chans.sort())
+    with tempfile.TemporaryDirectory() as temp_dir:
+        fovs, chans, imgs = test_utils.gen_fov_chan_names(1, 2, return_imgs=True)
```

Comment:
should add arg names here

Reply:
just to clarify, add arg names in `gen_fov_chan_names` or in `create_paired_xarray_fovs` ? or both now that I mention it?

Reply:
If the argument is a variable with an informative name, then it's okay to not use arg names. But when it's just raw ints we definitely should have them